provider.go

```diff
@@ -19,37 +22,100 @@ type providerModule struct {
 }
 
 func (m providerModule) Register(g *gin.RouterGroup) {
-	providers := g.Group("/providers")
-	{
-		providers.GET("", m.getProviderListHandler)
-		providers.POST("/search", m.searchProviderListHandler)
-	}
-
 	provider := g.Group("/provider")
 	{
-		provider.GET("/:providerId", m.getProviderHandler)
+		provider.GET("", m.getProviderListHandler)
 		provider.POST("", m.createProviderHandler)
-		provider.PUT("/:providerId", m.updateProviderHandler)
+
+		providerByID := provider.Group("/:providerId")
+		{
+			providerByID.GET("", m.getProviderHandler)
+			providerByID.PUT("", m.updateProviderHandler)
+		}
 	}
 }
 
-// getProviderListHandler godoc
+var providerJSONToColumns = map[string]string{
+	response.ProviderJSONFields.ProviderID: database.ProviderColumns.ProviderID,
+	response.ProviderJSONFields.Name:       database.ProviderColumns.Name,
+	response.ProviderJSONFields.URL:        database.ProviderColumns.URL,
+}
+
+var defaultGetProvidersOrderBy = orderby.Column{Name: database.ProviderColumns.ProviderID, Direction: orderby.Desc}
+
+// getBuildListHandler godoc
 // @id getProviderList
-// @summary Returns first 100 providers
+// @summary Get slice of providers.
+// @description List all providers, or a window of providers using the `limit` and `offset` query parameters. Allows optional filtering parameters.
+// @description Verbatim filters will match on the entire string used to find exact matches,
+// @description while the matching filters are meant for searches by humans where it tries to find soft matches and is therefore inaccurate by nature.
 // @tags provider
-// @success 200 {object} []response.Provider
-// @failure 401 {object} problem.Response "Unauthorized or missing jwt token"
+// @param limit query int false "Number of results to return. No limit if unset or non-positive. Required if `offset` is used." default(100)
```

Comment:
Dupe of previous.

Description here doesn't seem to match the changelog description:
> By default all these new endpoints use a default limit of 100, but this can
  be disabled by specifying `?limit=0`.

Assuming the changelog description is the actual behavior?
```suggestion
// @param limit query int false "Number of results to return. Set to 0 or non-positive to disable. Required if `offset` is used." default(100)
```
Alt.
```suggestion
// @param limit query int false "Number of results to return. Set to 0 to disable. Required if `offset` is used." minimum(0) default(100)
```